kerastuner/engine/multi_execution_tuner.py

```diff
@@ -83,7 +83,7 @@ def run_trial(self, trial, *fit_args, **fit_kwargs):
         # Run the training process multiple times.
         metrics = collections.defaultdict(list)
         for execution in range(self.executions_per_trial):
-            model = self._build_model(trial.hyperparameters.copy())
+            model = self._build_model(trial.hyperparameters)
```

Comment:
Isn't it strange that `trial.hyperparameters` would be out of sync with `oracle.hyperparameters`?

Reply:
The only time these would be out of sync is if `allow_new_entries=True` and `tune_new_entries=False`. To me, it makes sense in that case to have the Trial show that it is using the default values for these HPs

This implementation also allows us to take care of `oracle.update_space` for users, by putting it in the `on_trial_end` hook. This way any HPs added during `run_trial` (even HPs for things like preprocessing) will be picked up by the Oracle